modules/custom/az_core/az_core.module

```diff
@@ -100,3 +100,61 @@ function az_core_block_access(Block $block, $operation, AccountInterface $accoun
   }
   return AccessResult::neutral();
 }
+
+/**
+ * Implements hook_token_info().
+ */
+function az_core_token_info() {
+  return [
+    'tokens' => [
+      'node' => [
+        'az-canonical-url' => [
+          'name' => t("Canonical URL"),
+          'description' => t("Switches between field_az_link and node:url."),
+        ],
+      ],
+    ],
+  ];
+}
+
+/**
+ * Implements hook_token().
+ */
+function az_core_tokens($type, $tokens, array $data, array $options, \Drupal\Core\Render\BubbleableMetadata $bubbleable_metadata) {
+  $replacements = [];
+  if ($type == 'node') {
+    foreach ($tokens as $name => $original) {
+      // Find the desired token by name.
+      switch ($name) {
+        case 'az-canonical-url':
+          $replacements[$original] = az_core_canonical_url($data);
+          break;
+      }
+    }
+  }
+
+    return $replacements;
+}
+
+/**
+ * @TODO internal paths don't work.
+ */
+function az_core_canonical_url($data) {
+  /** @var \Drupal\node\NodeInterface $node */
+  $node = $data['node'];
+  $options = ['absolute' => TRUE];
+  $url = \Drupal\Core\Url::fromRoute('entity.node.canonical', ['node' => $node->id()], $options);
+  $default_url = $url->toString();
+  $url = $default_url;
+
+  if (!empty($node->field_az_link[0]->uri)){
+    try {
+      $url = Url::fromUri($node->field_az_link[0]->uri);
+      $url = $url->toString();
+    } catch (\InvalidArgumentException $e) {
+      $url = $default_url;
+    }
+  }
+
+  return $url;
```

Comment:
Since the field formatter provided by Field Group Link is [actually expecting a URI](https://git.drupalcode.org/project/field_group_link/-/blob/8.x-3.x/src/Plugin/field_group/FieldGroupFormatter/Link.php#L205-211) that it will generate a URL from, I think we might want to do something like this here:
```suggestion
  $url = \Drupal\Core\Url::fromRoute('entity.node.canonical', ['node' => $node->id()]);
  $uri = $url->getUri();

  if (!empty($node->field_az_link[0]->uri)) {
    $uri = $node->field_az_link[0]->uri;
  }

  return $uri;
```

Reply:
@tadean Does this change seem reasonable to you?

Reply:
@joeparsons Yes, looks good to me. The field group does seem to do its own exception handling so the try/catch shouldn't be needed here. This is much cleaner.